### PR TITLE
feat(deployment): replicate broken state sync in docker upgrade test

### PR DESCRIPTION
**Pull request by mhofman**
refs: #8081

## Description

Replicates the state of mainnet validators in the docker upgrade test to prepare for upcoming fix.

### Security Considerations

None

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

This is an integration test capturing the current state of the public chain

### Upgrade Considerations

Will help validate fix to be included in next upgrade

### CODE REVIEW

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh [hunk 1]
```diff
@@ -174,6 +174,16 @@ OFFER=$(mktemp -t agops.XXX)
...
-# # TODO test bidding
-# # TODO liquidations
-# # agops inter bid by-price --price 1 --give 1.0IST  --from $GOV1ADDR --keyring-backend test
+# replicate state-sync of node
+# this will cause the swing-store to prune some data
+killAgd
+EXPORT_DIR=$(mktemp -t -d swing-store-export-upgrade-10-XXX)
+make_swing_store_snapshot $EXPORT_DIR || fail "Couldn't make swing-store snapshot"
+test_val "$(compare_swing_store_export_data $EXPORT_DIR)" "match" "swing-store export data"
+restore_swing_store_snapshot $EXPORT_DIR || fail "Couldn't restore swing-store snapshot"
```
Comment (warner) [inline packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh:183]: Does this overwrite the previous `.agoric` directory with the one from state sync? Seems like a reasonable thing to do, just wanted to confirm.

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh [hunk 2]
```diff
@@ -174,6 +174,16 @@ OFFER=$(mktemp -t agops.XXX)
...
+rm -rf $EXPORT_DIR
+startAgd
+
+# # TODO fully test bidding
+# # TODO test liquidations
+agops inter bid by-price --price 1 --give 1.0IST  --from $GOV1ADDR --keyring-backend test
```
Comment (raphdev) [inline packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh:189]: This wasn't enabled. Do you intend to add a bid?

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh [hunk 3]
```diff
@@ -106,3 +106,50 @@ submitDeliverInbound() {
...
+  mv "$EXPORT_DATA_FILE" "$EXPORT_DATA_UNTRUSTED_FILE"
+  agd export --height $EXPORT_HEIGHT | jq -cr '.app_state.vstorage.data[] | if .path | startswith("swingStore.") then [.path[11:],.value] else empty end' > "$EXPORT_DATA_FILE"
+
+  jq -n "$EXPORT_MANIFEST | .untrustedData=\"$(basename -- "$EXPORT_DATA_UNTRUSTED_FILE")\"" > "$EXPORT_MANIFEST_FILE"
+
+  echo "Successful swing-store export for block $EXPORT_HEIGHT"
+)}
+
+restore_swing_store_snapshot() {( set -euo pipefail
+  rm -f $HOME/.agoric/data/agoric/swingstore.sqlite
```
Comment (warner) [inline packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh:129]: oh, it only overwrites the swingset DB, not the rest of cosmos.. that doesn't resemble anything that real validators would do, should we have this step do something (maybe involving a genesis export/import) to simulate what happens to the IAVL tree too?

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh [hunk 4]
```diff
@@ -106,3 +106,50 @@ submitDeliverInbound() {
...
+    echo "missing-export-data"
+    return
+  fi
+  
+  if [ -z "$EXPORT_DATA_UNTRUSTED_FILE" ]; then
+    echo "missing-untrusted-export-data"
+    return
+  fi
+  
+  diff <(cat "$EXPORT_DIR/$EXPORT_DATA_FILE" | sort) <(cat "$EXPORT_DIR/$EXPORT_DATA_UNTRUSTED_FILE" | sort) >&2 && {
```
Comment (warner) [inline packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh:150]: FWIW you can get `jq` to do the sort for you, `--sort-keys`, might be nicer than doing the sort here

**Review by warner — COMMENTED** [review]
sounds reasonable to me, but I'll defer to folks who know this test better

**Review by raphdev — APPROVED** [review]
This LGTM, the agd functions are a welcome improvement. Only nit is enabling a command that was previously commented out as part of a TODO, but doesn't seem consequential.

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh (hunk 2 above)
Comment (mhofman) [reply packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh:189]: yes

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh (hunk 1 above)
Comment (mhofman) [reply packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/actions.sh:183]: This simulates a state-sync restore. To do that it deletes `~/.agoric/data/agoric`

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh (hunk 3 above)
Comment (mhofman) [reply packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh:129]: I'm assuming the rest of the cosmos DB is restored to the same state. This only simulates the swing-store state restore.

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh (hunk 4 above)
Comment (mhofman) [reply packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-10/env_setup.sh:150]: I wanted to keep the order of keys intact in the files

**Pull request merged**